### ISSUE TITLE
enable option to add sample photo to walk data entry

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/UKBMS-online#282]
e.g. to form like https://ukbms.org/input-data?location_type_id=777&sample_id=5624248

the ebms equivalent form already has this functionality, e.g. https://butterfly-monitoring.net/walk-data-entry?sample_id=18800483

@JimBacon can you estimate the time to do this

**Comment by JimBacon** [comment]
Superficially, adding 'Transect level images' appears to be a matter of adding a configuration option and ticking a box which would take a minute or two (having spent an hour or two trying to find the option).

However, the prebuilt form, 'ukbms_sectioned_transects_input_sample_2', which is the form in question, is different on EBMS, UKBMS and in the client_helpers repository. One of the differences is in the configuration of the photo upload control. It might be a good opportunity to try to straighten it all out and return to a single code base which would be easier to maintain in the long run. I expect that might take a few days.

**Comment by JimBacon** [comment]
Also, requested by email, was the option of adding photos of occurrences. Again, this is superficially a few minutes work to add a configuration option and tick a box. The form doesn't work as expected, however, so it needs some poking around to explain that. Probably not a big issue but allow a day for resolving, committing changes and making available.

**Comment by IanMiddlebrook** [comment]
I think there could be two routes for photos:

At the walk/sample level, you would be able to upload photos from the walk - habitat/management related. You could also add photos of any unusual butterfly/sighting. Although this would not then be associated with the relevant occurrence, it could still be useful. For example, if I'm checking out an unusual sighting, I often go to that walk/sample page to see if they've added any note/comment about it - I could also see if they've uploaded any photos.

Alternatively, the species photos could be linked to the occurrence to aid verification in iRecord. At the moment, whilst I can see my UKBMS records in iRecord as a User, I'm not able to add photos to them, even through iRecord. Turning that option on in iRecord would be a start, but making some of that iRecord functionality available on the UKBMS website would avoid the need to switch between the two.

**Comment by JimBacon** [comment]
Agreed. The title of the issue and my first comment correspond with your first point about adding photos at the walk/sample level. My second comment corresponds to your second point of photos linked to occurrences.

You could have either one or both options available.

**Comment by DavidRoy** [comment]
Let's go for the sample-level photos to start with.  I'll raise a separate issue for adding photos to occurrences but we will want that functionality. Jim - I can't remember if this is already possible for the ebms version of the walk-data-entry-2 form
